### ISSUE TITLE
curried create returns object of wrong type

### ISSUE THREAD

**Issue by dporf** [unadlib/mutative#57]
mutative 1.0.8
`create` returns `T` instead of `Draft<T>`
```ts
import { create } from 'mutative';
import { DeepReadonly } from 'ts-essentials';

type A = {
  x: string;
};

type B = {
  a: A;
};

const b: DeepReadonly<B> = { a: { x: 'asd' } };

const [draft, finalize] = create(b);

draft.a.x = 'dsa'; //Cannot assign to 'x' because it is a read-only property.

console.log(finalize());
```

**Comment by unadlib** [comment]
hi @dporf , I have reproduced the issue.

Thanks for reporting it. I will fix it later.

**Comment by unadlib** [comment]
This issue has been fixed in v1.0.9. Feel free to use it.
